src/table.cpp

```diff
@@ -97,3 +98,38 @@ std::vector<JoinResult> Table::hashJoin(size_t index, Table &other, size_t other
 
     return result;
 }
+
+std::vector<JoinResult> Table::hashJoinRows(size_t index, const std::vector<std::size_t> &rows, Table &other, size_t other_index) {
+    std::vector<JoinResult> result;
+
+    auto &column = this->column(index)->data();
+    auto &other_column = other.column(other_index)->data();
+
+    // We weren't using multimaps before, so actually the hash join operation
+    // wasn't implemented correctly. Definitely need them for Task 3, because we have multiple
+    // rows with the same values
+    using map_t = std::unordered_multimap<uint32_t, size_t>;
```

Comment:
I was testing with `unordered_multimaps` before, but couldn't fix the occurring segfaults and allocation failures.

Because there are only unique values in the original join benchmark, I decided not to fix this back then.